gateway/handle.go

```diff
@@ -763,14 +766,14 @@ func (gw *Handle) extractJobsFromInternalBatchPayload(reqType string, body []byt
 	if err != nil {
 		stat.RequestFailed(response.InvalidJSON)
 		stat.Report(gw.stats)
-		return nil, errors.New(response.InvalidJSON)
+		return nil, response.InvalidJSON, fmt.Errorf("unmarshalling request body: %w", err)
```

Comment:
This is not an elegant way to do this. Using wrapping and using `if errors.As(err, &e)` to extract response error would be a better approach. https://go.dev/blog/go1.13-errors

If this is too complex given the time constrains, I will log the error here.